src/libraries/System.Net.Sockets/tests/FunctionalTests/SocketTestHelper.cs

```diff
@@ -179,6 +180,33 @@ public override Task<(Socket socket, byte[] buffer)> AcceptAsync(Socket s, int r
             s.SendToAsync(buffer, SocketFlags.None, endPoint);
     }
 
+    // Same as above, but call the CancellationToken overloads where possible
```

Comment:
Maybe change these to use a cancelable token?
```C#
private readonly CancellationTokenSource _cts = new CancellationTokenSource();
...
        public override Task ConnectAsync(Socket s, EndPoint endPoint) =>
            s.ConnectAsync(endPoint, _cts.Token).AsTask();
```
?

Reply:
Good idea, will change.